### PR TITLE
replace network sdk for spring

### PR DESCRIPTION
**Pull request by AllyW**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az spring create

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
network bump

File: src/spring/azext_spring/_validators.py [hunk 1]
```diff
@@ -358,51 +358,74 @@ def validate_vnet(cmd, namespace):
...
+            args_schema.subscription_id = AAZStrArg(
+                options=['--subscription-id'],
+            )
+            return args_schema
+
+        def pre_operations(self):
+            from azure.cli.core.aaz import has_value
+            args = self.ctx.args
+            if has_value(args.subscription_id):
+                self.ctx.subscription_id = args.subscription_id
```
Comment (kairu-ms) [inline src/spring/azext_spring/_validators.py:413]: I think it's an internal property `_subscription_id`.

```suggestion
                self.ctx._subscription_id = args.subscription_id
```

File: src/spring/azext_spring/_validators.py [hunk 2]
```diff
@@ -358,51 +358,74 @@ def validate_vnet(cmd, namespace):
...
+    # return network_client.virtual_networks.get(vnet['resource_group'], vnet['resource_name'])
+    from .aaz.latest.network.vnet import Show as _VirtualNetworkShow
+
+    class VirtualNetworkShow(_VirtualNetworkShow):
+        @classmethod
+        def _build_arguments_schema(cls, *args, **kwargs):
+            from azure.cli.core.aaz import AAZStrArg
+            args_schema = super()._build_arguments_schema(*args, **kwargs)
+            args_schema.subscription_id = AAZStrArg(
+                options=['--subscription-id'],
```
Comment (kairu-ms) [inline src/spring/azext_spring/_validators.py:405]: It can be removed because it's not exposed to users.
```suggestion
```

**Review by kairu-ms — APPROVED** [review]

File: src/spring/azext_spring/_validators.py [hunk 3]
```diff
@@ -358,51 +358,61 @@ def validate_vnet(cmd, namespace):
...
+            from azure.cli.core.aaz import AAZStrArg
+            args_schema = super()._build_arguments_schema(*args, **kwargs)
+            args_schema.subscription_id = AAZStrArg()
+            return args_schema
+
+        def pre_operations(self):
+            from azure.cli.core.aaz import has_value
+            args = self.ctx.args
+            if has_value(args.subscription_id):
+                self.ctx._subscription_id = args.subscription_id
```
Comment (kairu-ms) [inline src/spring/azext_spring/_validators.py:409]:
```
self.ctx._subscription_id = args.subscription_id.to_serialized_data()
```

**Pull request merged**